Apidaze.SDK.Tests.Integration/XML/Program.cs

```diff
@@ -0,0 +1,170 @@
+            {
+                _keepGoing = false;
+                lock (Listener)
+                {
+                    Listener.Stop();
+                }
+                try
+                {
+                    _mainLoop.Wait();
+                }
+                catch { }
+            }
+
+            /// <summary>
+            /// The main loop to handle requests into the HttpListener
+            /// </summary>
+            /// <returns></returns>
+            private static async Task MainLoop()
+            {
+
+                Listener.Prefixes.Add(LOCALHOST + INTRO);
+
+                Listener.Start();
+                while (_keepGoing)
+                {
+                    try
+                    {
+
+                        var context = await Listener.GetContextAsync();
+                        lock (Listener)
+                        {
+                            if (_keepGoing) ProcessRequest(context);
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        Console.WriteLine(e.Message);
+                        if (e is HttpListenerException) return;
+                    }
+                }
+            }
+
+            /// <summary>
+            /// Handle an incoming request
+            /// </summary>
+            /// <param name="context">The context of the incoming request</param>
+            private static void ProcessRequest(HttpListenerContext context)
+            {
+                using (var response = context.Response)
+                {
+                    try
+                    {
+                        var handled = false;
+                        switch (context.Request.Url.AbsolutePath)
+                        {
+
+                            case "/":
+                                handled = GetResponse(context, response, handled);
+                                break;
+                        }
+                        if (!handled)
+                        {
+                            response.StatusCode = 404;
+                        }
+
+                        Console.WriteLine(response.ContentType + response.StatusDescription);
+                    }
+                    catch (Exception e)
+                    {
+                        //Return the exception details the client - you may or may not want to do this
+                        response.StatusCode = 500;
+                        response.ContentType = "text/plain";
+                        var buffer = Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(e));
+                        response.ContentLength64 = buffer.Length;
+                        response.OutputStream.Write(buffer, 0, buffer.Length);
+
+                        //TODO: Log the exception
+                    }
+                }
+            }
+
+            private static bool GetResponse(HttpListenerContext context, HttpListenerResponse response, bool handled)
```

Comment:
The handled parameter is also here not necessary. Also from a logical standpoint, you're evaluating the request http method in a function that has actually nothing to do with the request itself. This method should IMHO, just return the XML and not care about anything related to the http request. The request itself should be evaluated in the `ProcessRequest(HttpListenerContext context)` method. That way, you check for the method just once, and not in every function that returns an XML doc.

Reply:
Please see my last commit. It's fixed.